File: chapter_01/BinarySearch.java
public class BinarySearch {
    public static void main(String... args) {
        int[] array = {1, 6, 8, 20, 53, 65, 77, 87, 95};

        System.out.println(search(array, 65));
    }

    private static int search(int[] array, int item) {
        int low = 0;
        int high = array.length;

        while(low <= high) {
            int mid = (low + high) / 2;
            int guess = array[mid];
            
            if(guess == item) {
                return mid;
            };
            if(guess > item) {
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }

        return item;
    }
}

File: chapter_01/binary_search.php
<?php

$arr = [1, 6, 8, 20, 53, 65, 77, 87, 95];

function binary_search($arr, $item) {
    $low = 0;
    $high = count($arr) - 1;
    
    while($low <= $high) {
        $mid = floor(($low + $high) / 2);
        $guess = $arr[$mid];

        if($guess === $item) {
            return $mid;
        } else if($guess > $item) {
            $high = $mid - 1;
        } else {
            $low = $mid + 1;
        }
    }

    return null;
}


echo binary_search($arr, 65);